src/daft-plan/src/logical_ops/actor_pool_project.rs

```diff
@@ -30,7 +30,7 @@ pub struct ActorPoolProject {
 impl ActorPoolProject {
     pub(crate) fn try_new(input: Arc<LogicalPlan>, projection: Vec<ExprRef>) -> Result<Self> {
         let (projection, fields) =
-            resolve_exprs(projection, input.schema().as_ref()).context(CreationSnafu)?;
+            resolve_exprs(projection, input.schema().as_ref(), true).context(CreationSnafu)?;
```

Comment:
Not important now, but as this becomes more complex we might consider maybe a builder `ExpressionResolver` API that has things like `.with_schema()`, `.allow_stateful_udfs()` etc.

Each LogicalOp can then implement their own ExpressionResolver which can do their own thing

Reply:
yeah a builder would honestly be kinda nice